wagtailio/api.py

```diff
@@ -1,8 +1,24 @@
 from wagtail.api.v2.router import WagtailAPIRouter
 from wagtail.api.v2.views import PagesAPIViewSet
-from wagtail.images.api.v2.views import ImagesAPIViewSet
+from wagtail.core.models import Page
+
+from wagtailio.areweheadlessyet.models import AreWeHeadlessYetHomePage
 
 api_router = WagtailAPIRouter("wagtailapi")
 
-api_router.register_endpoint("pages", PagesAPIViewSet)
-api_router.register_endpoint("images", ImagesAPIViewSet)
+
+class AreWeHeadlessYetPagesAPIViewSet(PagesAPIViewSet):
+    def get_base_queryset(self):
+        """Returns a queryset containing only pages from the AreWeHeadLessYet site."""
+
+        areweheadlessyet_queryset = AreWeHeadlessYetHomePage.objects.all()
+        if not areweheadlessyet_queryset:
+            return areweheadlessyet_queryset
+
+        areweheadlessyet_root_page = areweheadlessyet_queryset[0]
```

Comment:
You could use ``.first()`` here which will return ``None`` if the queryset is empty so you wouldn't need to perform the query above to check if there are any pages in the queryset first:

```python
areweheadlessyet_root_page = areweheadlessyet_queryset.first()

if areweheadlessyet_root_page:
    return Page.objects.live().descendant_of(
        areweheadlessyet_root_page, inclusive=True
    )
else:
    return Page.objects.none()
```